### PR TITLE
[Compute] New-AzRestorePoint and New-AzRestorePointCollection

### PR DESCRIPTION
**Pull request by haagha**
(https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1098)

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by haagha** [comment]
@BethanyZhou can you please add this to the March release? Thanks.

File: src/Compute/Compute/RestorePoints/NewAzRestorePoint.cs [hunk 1]
```diff
@@ -39,20 +39,29 @@ public class NewAzureRestorePoint : ComputeAutomationBaseCmdlet
...
 
         [Parameter(
-            Position = 1,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true)]
         [Alias("RestorePointName")]
         public string Name { get; set; }
 
         [Parameter(
-            Position = 3,
```
Comment (BethanyZhou) [inline src/Compute/Compute/RestorePoints/NewAzRestorePoint.cs:55]: Removing position may break user's script, please revert these changes.

File: src/Compute/Compute/ChangeLog.md [hunk 2]
```diff
@@ -25,6 +25,8 @@
 * Updated `Get-AzVm` to include `GetVirtualMachineById` parameter set.
 * Edited the documentation for the cmdlet `Set-AzVMADDomainExtension` to ensure the example is accurate. 
 * Improved description and examples for disk creation.
+* Adding new parameters to 'New-AzRestorePoint' and 'New-AzRestorePointCollection' for copying Restore Points and Restore Point Collections.
+* Adding Zone and PlacementGroupId Parameters to Repair-AzVmssServiceFabricUpdateDomain.
```
Comment (BethanyZhou) [inline src/Compute/Compute/ChangeLog.md:29]:
```suggestion
* Added new parameters to `New-AzRestorePoint` and `New-AzRestorePointCollection` for copying Restore Points and Restore Point Collections.
* Added `Zone` and `PlacementGroupId` Parameters to `Repair-AzVmssServiceFabricUpdateDomain`.
```

**Comment by BethanyZhou** [comment]
Please fix analyze issues and test issues.

  [xUnit.net 00:01:50.05]     Microsoft.Azure.Commands.Compute.Test.ScenarioTests.RestorePointsTests.TestRestorePoints [FAIL]

 /Users/runner/work/1/s/artifacts//StaticAnalysisResults/BreakingChangeIssues.csv Errors
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.Compute","Microsoft.Azure.Commands.Compute.Automation.NewAzureRestorePoint","New-AzRestorePoint","0","1050","The parameter set '__AllParameterSets' for cmdlet 'New-AzRestorePoint' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'New-AzRestorePoint'."
  "Az.Compute","Microsoft.Azure.Commands.Compute.Automation.NewAzureRestorePointCollection","New-AzRestorePointCollection","0","1050","The parameter set '__AllParameterSets' for cmdlet 'New-AzRestorePointCollection' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'New-AzRestorePointCollection'."

**Comment by BethanyZhou** [comment]
> @BethanyZhou can you please add this to the March release? Thanks.

Our code freeze date is 03/28 and we are building release candidate for March release. 

Any special reason to must have this PR in March release?  If so, please fix comments above ASAP and let us see this PR whether has change to squeeze into release candidate.

**Comment by haagha** [comment]
@BethanyZhou yes this is request customers are waiting on, I will fix the PR and let you know.

**Comment by haagha** [comment]
@BethanyZhou The Static analysis failure is a false positive. 
It says in this document https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md
"Note: Sometimes the error listed in the .csv file can be a false positive (for example, if you change a parameter attribute to span all parameter sets rather than individual parameter sets)."

**Comment by BethanyZhou** [comment]
Yes, please suppress them in suppress the breaking change errors in tools/StaticAnalysis/Exceptions/${ModuleName}/BreakingChangeIssues.csv"

File: src/Compute/Compute/help/New-AzRestorePoint.md [hunk 3]
```diff
@@ -13,9 +13,9 @@ This cmdlet can create a New Restore Point
 ## SYNTAX
 
 ```
-New-AzRestorePoint [-ResourceGroupName] <String> [-RestorePointCollectionName] <String> [-Name] <String>
- [[-DisksToExclude] <String[]>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
- [<CommonParameters>]
+New-AzRestorePoint [-ResourceGroupName] <String> -RestorePointCollectionName <String> -Name <String>
+ [-Location <String>] -RestorePointId <String> [-DisksToExclude <String[]>]
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzRestorePoint.md:17]: please update markdown files to keep sync with code implementation. The syntax says RestorePointCollectionName is not a positional parameter now. Need be updated.

File: src/Compute/Compute/help/New-AzRestorePoint.md [hunk 4]
```diff
@@ -51,27 +51,42 @@ Accept wildcard characters: False
...
 Required: False
-Position: 3
+Position: Named
 Default value: None
 Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -Location
+{{ Fill Location Description }}
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzRestorePoint.md:66]: The description of Location is missing, please add help message for it.

File: src/Compute/Compute/RestorePoints/NewAzRestorePoint.cs [hunk 5]
```diff
@@ -53,6 +53,18 @@ public class NewAzureRestorePoint : ComputeAutomationBaseCmdlet
 
         [Parameter(
             Position = 3,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false)]
+        public string Location { get; set; }
```
Comment (BethanyZhou) [inline src/Compute/Compute/RestorePoints/NewAzRestorePoint.cs:58]: please add help message for new added parameters.

File: src/Compute/Compute/help/New-AzRestorePoint.md [hunk 6]
```diff
@@ -96,22 +111,37 @@ Accept wildcard characters: False
...
 Required: True
-Position: 1
+Position: Named
 Default value: None
 Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -RestorePointId
+{{ Fill RestorePointId Description }}
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzRestorePoint.md:126]: This one needs help message as well.

File: src/Compute/Compute/help/New-AzRestorePointCollection.md [hunk 7]
```diff
@@ -91,16 +110,31 @@ Accept pipeline input: True (ByPropertyName)
...
 
 ```yaml
-Type: String
-Parameter Sets: (All)
+Type: System.String
+Parameter Sets: DefaultParameter
 Aliases:
 
 Required: True
-Position: 2
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/New-AzRestorePointCollection.md:103]: Please update markdown files.

File: src/Compute/Compute/help/Repair-AzVmssServiceFabricUpdateDomain.md [hunk 8]
```diff
@@ -166,6 +182,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -Zone
+{{ Fill Zone Description }}
```
Comment (BethanyZhou) [inline src/Compute/Compute/help/Repair-AzVmssServiceFabricUpdateDomain.md:186]: Needs description. To add it, please add help message in cs file.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/help/New-AzRestorePointCollection.md [hunk 9]
```diff
@@ -91,16 +110,31 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -RestorePointCollectionId
+{{ Fill RestorePointCollectionId Description }}
```
Comment (dingmeng-xue) [inline src/Compute/Compute/help/New-AzRestorePointCollection.md:114]: Description is empty.

**Comment by haagha** [comment]
Added static analysis to breaking changes supression csv and updated descriptions and help files.

**Comment by BethanyZhou** [comment]
Seems like you forget to update Repair-AzVmssServiceFabricUpdateDomain.md the description of Zone and PlacementGroupId  is empty, please have a look.

**Review by BethanyZhou — COMMENTED** [review]
All look good to me except Repair-AzVmssServiceFabricUpdateDomain.md

**Review by BethanyZhou — APPROVED** [review]
Looks good to me
